### PR TITLE
Apply serialization proxy pattern to BattleRecord and BattleRecords

### PR DESCRIPTION
**Pull request by DanVanAtta**
Based on #945

This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/947)

### CODE REVIEW

File: src/games/strategy/triplea/delegate/dataObjects/BattleRecord.java [hunk 1]
```diff
@@ -33,67 +35,108 @@
     BLITZED, CONQUERED, WON_WITHOUT_CONQUERING, WON_WITH_ENEMY_LEFT, STALEMATE, LOST, BOMBED, AIR_BATTLE_WON, AIR_BATTLE_LOST, AIR_BATTLE_STALEMATE, NO_BATTLE
   }
 
+
   private static final long serialVersionUID = 3642216371483289106L;
-  private Territory m_battleSite;
-  private PlayerID m_attacker;
+  private Territory battleSite;
+  private PlayerID attacker;
   private PlayerID m_defender;
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/delegate/dataObjects/BattleRecord.java]: Why not remove m_ here?

**Comment by ron-murhammer** [comment]
@DanVanAtta I'm pretty sure there are some issues with these changes. I did a quick test of just launching a map and it fails immediately. Please test this and here is the exception:

Exception: games.strategy.triplea.delegate.dataObjects.BattleRecords$SerializationProxy
java.io.NotSerializableException: games.strategy.triplea.delegate.dataObjects.BattleRecords$SerializationProxy
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
	at games.strategy.engine.framework.GameDataManager.writeDelegates(GameDataManager.java:247)

**Comment by DanVanAtta** [comment]
I think there was a problem with how this was split apart from the AllianceProxy change. Try again @ron-murhammer , I can no longer repro the serialization error.

**Comment by DanVanAtta** [comment]
Probably does not make sense to merge this after 1.9, save game compatibility would be broken right away.

**Pull request merged**